pkg/controller/controller.go

```diff
@@ -1035,6 +1038,18 @@ func (p *csiProvisioner) Delete(volume *v1.PersistentVolume) error {
 	ctx, cancel := context.WithTimeout(context.Background(), p.timeout)
 	defer cancel()
 
+	// Verify if volume is attached to a node before proceeding with deletion
+	vaList, err := p.vaLister.List(labels.Everything())
```

Comment:
VolumeAttachment objects will be in thousands in a scaled environment. Listing all VA objects is quite expensive. Can we avoid such expensive call to API server?

Reply:
`vaLister` lists VolumeAttachments from in-memory informer cache. "Thousands" should be ok, "millions" could be problematic.

Reply:
Does informer limit how many objects are in the cache? How are we sure that all VolumeAttachment objects will fit into memory? If the lister ends up hitting the API server then this will introduce a performance regression, hence I'm a bit concerned with this code.